freemindlatex/compilation_client_lib.py

```diff
@@ -71,29 +71,35 @@ def CheckHealthy(self):
     return (response.status ==
             compilation_service_pb2.HealthCheckResponse.SERVING)
 
-  def CompileDir(self, directory):
+  def CompileDir(self, directory, mode):
     """Compiles the files in user's directory, and update the pdf file.
 
     The function will prepare the directory content, send it over for
     compilation. When there is a latex compilation error, we will put
     the latex error log at latex.log (or anything else specified by
     latex_error_log_filename).
 
-    Returns: boolean indicating if the compilation was successful.
-      When unceccessful, leaves log files.
-
     Args:
       directory: directory where user's files locate
+      mode: the way to compile,
+        e.g. compilation_service_pb2.LatexCompilationRequest.BEAMER
+
+    Returns: boolean indicating if the compilation was successful.
+      When unceccessful, leaves log files.
     """
 
-    target_pdf_loc = os.path.join(directory, 'slides.pdf')
     filename_and_mtime_list = GetMTimeListForDir(directory)
     compilation_request = compilation_service_pb2.LatexCompilationRequest()
     for filename, _ in filename_and_mtime_list:
       with open(os.path.join(directory, filename)) as infile:
         new_file_info = compilation_request.file_infos.add()
         new_file_info.filepath = filename
         new_file_info.content = infile.read()
+    compilation_request.compilation_mode = mode
+    if mode == compilation_service_pb2.LatexCompilationRequest.BEAMER:
```

Comment:
This type of mapping has occurred many times, which is often not very necessary. During compilation, we can be agnostic of the filename. Also when preparing the final document for users, we should be using the folder's name as the doc name, instead of just "slides" or "report". This makes it easier to look up for the files later on.